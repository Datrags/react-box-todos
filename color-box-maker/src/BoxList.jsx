import { useState } from 'react'
import Box from "./Box"
import NewBoxForm from "./NewBoxForm"
const BoxList = () => {

    const [boxes, setBoxes] = useState([]);
    const createBox = box => {
        setBoxes(boxes => [...boxes, box]);
    }

    const removeBox = id => {
        setBoxes(boxes => boxes.filter(box => box.id !== id));
    }


    const boxComponents = boxes.map(box => (
        <Box
            key={box.id}
            id={box.id}
            width={box.width}
            height={box.height}
            remove={removeBox}
            color={box.color} 
        />
    ))

return (
  <div>

    <NewBoxForm createBox={createBox}/>
    {boxComponents}
  </div>)

}

export default BoxList;