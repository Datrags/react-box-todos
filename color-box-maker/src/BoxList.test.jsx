import React from "react";
import { render, fireEvent} from "@testing-library/react";
import "@testing-library/jest-dom"
import BoxList from "./BoxList";

it("should render without crashing", () => {
    render(<BoxList></BoxList>);
});

it("should match snapshot", () => {
    const {asFragment} = render(<BoxList/>);
    expect(asFragment()).toMatchSnapshot();
});

it("should add a new box", () => {
    const boxList = render(<BoxList/>);
    fireEvent.click(boxList.getByText("Create New Box"));
    let remBtn = boxList.getByText("X");
    expect(remBtn).toBeInTheDocument();
})

it("should have a working delete feature", () => {
    const boxList = render(<BoxList/>);
    fireEvent.click(boxList.getByText("Create New Box"));
    let remBtn = boxList.getByText("X");
    expect(remBtn).toBeInTheDocument();
    fireEvent.click(remBtn)
    expect(remBtn).not.toBeInTheDocument();
})