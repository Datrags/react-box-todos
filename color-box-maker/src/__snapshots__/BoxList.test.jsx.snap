// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="height"
      >
        Height
      </label>
      <input
        id="height"
        name="height"
        placeholder="Height"
        type="text"
        value="100px"
      />
      <br />
      <label
        for="width"
      >
        Width
      </label>
      <input
        id="width"
        name="width"
        placeholder="Width"
        type="text"
        value="100px"
      />
      <br />
      <label
        for="color"
      >
        Color
      </label>
      <input
        id="color"
        name="color"
        placeholder="Color"
        type="text"
        value="#000"
      />
      <br />
      <button>
        Create New Box
      </button>
    </form>
  </div>
</DocumentFragment>
`;
